Allow overriding robots settings in news and events

dmolineus opened the pull request:
| Q                | A
| -----------------| ---
| Fixed issues     | Fixes #...
| Docs PR or issue | contao/docs#...

This pull request adds the ability to override robots setting in the news, events and faqs. If robots setting is overriden the robots tag is overriden and the item is excluded from xml sitemap if "noindex,nofollow" is chosen.

Known limitation: If a page is set to `noindex,nofollow` an item can't be included in the xml sitemap.

/cc @frontendschlampe

dmolineus committed:
Allow to override robots setting

If robots setting is overriden the robots tag is overriden and the
item is excluded from xml sitemap if "noindex,nofollow" is chosen

dmolineus commented:
Failing CI seems not to be related to this PR.

aschempp commented on calendar-bundle/src/Resources/contao/languages/en/tl_calendar_events.xlf:
```diff
@@ -56,6 +56,12 @@
+        <source>Robots tag</source>
+      </trans-unit>
+      <trans-unit id="tl_calendar_events.robots.1">
+        <source>Here you can override how search engines handle the page. If empty the page setting is used.</source>
```
Should this be _event_ instead of _page_?

dmolineus replied:
Resolved in e4056844494cffdfa1059e19d213deb6250ac083

aschempp commented on faq-bundle/src/Resources/contao/modules/ModuleFaq.php:
```diff
@@ -103,6 +103,11 @@ public function getSearchablePages($arrPages, $intRoot=0, $blnIsSitemap=false)
 				{
 					while ($objItems->next())
 					{
+						if ($blnIsSitemap && $objItems->robots === 'noindex,nofollow')
```
should this check the page settings as well?

dmolineus replied:
It's already checked above, so no need to check it again https://github.com/contao/contao/blob/449a6fbb3717af62b5f40f1f71cd9f5c71fb87f1/faq-bundle/src/Resources/contao/modules/ModuleFaq.php#L87

aschempp commented on calendar-bundle/src/Resources/contao/classes/Calendar.php:
```diff
@@ -373,6 +373,11 @@ public function getSearchablePages($arrPages, $intRoot=0, $blnIsSitemap=false)
 				{
 					while ($objEvents->next())
 					{
+						if ($blnIsSitemap && $objEvents->robots === 'noindex,nofollow')
```
should this check the page settings as well?

dmolineus replied:
It's already checked above, so no need to check it again

https://github.com/contao/contao/blob/449a6fbb3717af62b5f40f1f71cd9f5c71fb87f1/calendar-bundle/src/Resources/contao/classes/Calendar.php#L357

aschempp commented on news-bundle/src/Resources/contao/classes/News.php:
```diff
@@ -336,6 +336,11 @@ public function getSearchablePages($arrPages, $intRoot=0, $blnIsSitemap=false)
 				{
 					while ($objArticle->next())
 					{
+						if ($blnIsSitemap && $objArticle->robots === 'noindex,nofollow')
```
should this check the page settings as well?

dmolineus replied:
It's already checked above, so no need to check it again

https://github.com/contao/contao/blob/449a6fbb3717af62b5f40f1f71cd9f5c71fb87f1/news-bundle/src/Resources/contao/classes/News.php#L320

aschempp commented on calendar-bundle/src/Resources/contao/dca/tl_calendar_events.php:
```diff
@@ -117,7 +117,7 @@
 	(
 		'__selector__'                => array('source', 'addTime', 'addImage', 'recurring', 'addEnclosure', 'overwriteMeta'),
-		'default'                     => '{title_legend},title,alias,author;{date_legend},addTime,startDate,endDate;{source_legend:hide},source;{meta_legend},pageTitle,description,serpPreview;{details_legend},location,address,teaser;{image_legend},addImage;{recurring_legend},recurring;{enclosure_legend:hide},addEnclosure;{expert_legend:hide},cssClass,noComments;{publish_legend},published,start,stop',
+		'default'                     => '{title_legend},title,alias,author;{date_legend},addTime,startDate,endDate;{source_legend:hide},source;{meta_legend},pageTitle,robots,description,serpPreview;{details_legend},location,address,teaser;{image_legend},addImage;{recurring_legend},recurring;{enclosure_legend:hide},addEnclosure;{expert_legend:hide},cssClass,noComments;{publish_legend},published,start,s ...
```
I wonder why FAQ items do not have the other meta information, maybe it would make sense to add that as well (not sure if in that PR though)?

dmolineus replied:
I also wondered :-)

dmolineus replied:
Added another PR for it: https://github.com/contao/contao/pull/1779

aschempp reviewed (commented):
Looking good! Just a few minor thoughts 😊

dmolineus committed:
Improve translation

dmolineus commented:
If https://github.com/contao/contao/pull/1779 got accepted we might want to move the robots field to the meta legend for faqs as well.

dmolineus commented:
What are the chances that the feature will be included in Contao 4.10? Is there anything I can do further to get it shipped with the next release?

Toflar reviewed (dismissed):
Rebasing/Making sure the tests pass would be great.
Otherwise I'm +1 on 4.10 here.

leofeyer committed:
Fix a few minor issues

leofeyer reviewed (changes requested):
The PR looks good to me, however, can we please remove the FAQ related changes and add them to #1779? Then we have everything in one place.

dmolineus committed:
Drop implementation for faqs

It will be added to #1779, see as requested in
https://github.com/contao/contao/pull/1778#pullrequestreview-437324426

dmolineus commented:
> The PR looks good to me, however, can we please remove the FAQ related changes and add them to #1779? Then we have everything in one place.

Done with a03f1b3  and b350a6a .

dmolineus committed:
Drop leftover annotation

leofeyer commented:
Thank you @dmolineus.
